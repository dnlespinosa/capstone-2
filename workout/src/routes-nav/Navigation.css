.Navigation {
    background-color: white;
    border-bottom: 2px solid rgba(0, 0, 0, 0.125);
    color: black
  }




ul {
  list-style-type: none;
}

.list-item {
  display: inline-flex;
  margin: 0 10px;
}

  
  .nav-link.active {
    font-weight: bold;
    color: black;
  }