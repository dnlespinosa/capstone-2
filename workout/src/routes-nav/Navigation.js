import React, { useContext, useState } from 'react';
import { Link, NavLink } from 'react-router-dom';

const Navigation = ({ currentUser, logout }) => {

    return (
        <>
            <ul>
                <li>
                    <NavLink to='/'>
                        Home
                    </NavLink>
                </li>

                {!currentUser ? 
                    <div>
                        <li>
                            <NavLink to='/login'>
                                Login
                            </NavLink>
                        </li>
                        <li>
                            <NavLink to='/signup'>
                                Signup
                            </NavLink>
                        </li>                       
                        </div>:
                        <div>
                            <li>
                                <NavLink to='/browse'>
                                    Browse Programs
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to='/your-program'>
                                    Your Program
                                </NavLink>
                            </li>
                            <li>
                                <NavLink to='/' onClick={logout}>
                                    Logout
                                </NavLink>
                            </li>
                        </div>
                        }
                
                
                
            </ul>
        </>
    )
}

export default Navigation