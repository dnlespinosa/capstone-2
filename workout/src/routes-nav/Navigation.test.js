import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Navigation from "./Navigation";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <Navigation />
        </UserProvider>
      </MemoryRouter>,
  );
});

