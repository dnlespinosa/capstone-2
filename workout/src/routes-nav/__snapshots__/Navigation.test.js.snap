// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Navigation"
  >
    <div
      style="display: inline-flex; margin: 0px 10px;"
    >
      <h1
        class="Headline"
      >
        TrainTime
      </h1>
      <ul>
        <li
          class="list-item"
        >
          <a
            aria-current="page"
            class="list-item active"
            href="/"
          >
            Home
          </a>
        </li>
        <div>
          <li
            class="list-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
          </li>
          <li
            class="list-item"
          >
            <a
              class=""
              href="/signup"
            >
              Signup
            </a>
          </li>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
